src/pages/workspaces/hooks/useCloudEnvironmentPolling.ts

```diff
@@ -0,0 +1,98 @@
+import _ from 'lodash/fp';
+import { useEffect, useRef, useState } from 'react';
+import { getDiskAppType } from 'src/analysis/utils/app-utils';
+import { getConvertedRuntimeStatus, getCurrentRuntime } from 'src/analysis/utils/runtime-utils';
+import { Ajax } from 'src/libs/ajax';
+import { PersistentDisk } from 'src/libs/ajax/leonardo/models/disk-models';
+import { ListRuntimeItem } from 'src/libs/ajax/leonardo/models/runtime-models';
+import { withErrorIgnoring, withErrorReporting } from 'src/libs/error';
+import { InitializedWorkspaceWrapper as Workspace } from 'src/pages/workspaces/workspace/useWorkspace';
+
+export interface CloudEnvironmentDetails {
+  runtimes?: ListRuntimeItem[];
+  refreshRuntimes: (maybeStale?: boolean) => Promise<void>;
+  persistentDisks?: PersistentDisk[];
+  appDataDisks?: PersistentDisk[];
+}
+
+export const useCloudEnvironmentPolling = (
+  name: string,
+  namespace: string,
+  workspace?: Workspace
+): CloudEnvironmentDetails => {
+  const [controller, setController] = useState(new window.AbortController());
```

Comment:
The `useCancellation` hook only cancels the fetch when the component is unmounted (such as when the user navigates away from the page), because the effect in `useCancellation` has no dependencies. 
We want the call to be canceled any time we need to re-run the effect in this hook.

`useCancelable` doesn't seem to work correctly either: calling the `abort` function returned by `useCancelable` seems to result in unpredictable behavior, and was very finicky to use. 
I think this might have to do with some non-deterministic ordering of when effects are run.

Either way, setting up the abort controller directly in this hook resulted in a predictable, controllable behavior.